Dockerfile

```diff
@@ -1,64 +1,106 @@
 RUN set -x \
-    && export DEBIAN_FRONTEND=noninteractive \
-    && apt-get -y update \
-    && apt-get -y install \
+ && apk --no-cache add \
+    bash \
+    curl \
+    file \
+    git \
     libseccomp-dev \
     rsync
+
+# Dapper/Drone/CI environment
+FROM build AS dapper
+ENV DAPPER_ENV GODEBUG REPO TAG DRONE_TAG PAT_USERNAME PAT_TOKEN KUBERNETES_VERSION DOCKER_BUILDKIT ASSERT_STATIC ASSERT_BORING
+ENV DAPPER_OUTPUT ./dist ./bin ./build
+ENV DAPPER_DOCKER_SOCKET true
+ENV DAPPER_TARGET dapper
+ENV DAPPER_RUN_ARGS "-v rke2-pkg:/go/pkg -v rke2-cache:/root/.cache/go-build"
+RUN curl -sL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.27.0
+WORKDIR /source
+# End Dapper stuff
+
 # Shell used for debugging
-FROM build AS shell
+FROM dapper AS shell
 RUN set -x \
-    && export DEBIAN_FRONTEND=noninteractive \
-    && apt-get -y update \
-    && apt-get -y install \
-    bash \
+ && apk --no-cache add \
     bash-completion \
-    git \
     iptables \
     less \
-    libseccomp2 \
     psmisc \
     rsync \
-    seccomp \
     socat \
     sudo \
     vim
 RUN GO111MODULE=off GOBIN=/usr/local/bin go get github.com/go-delve/delve/cmd/dlv
-RUN cp -f /etc/skel/.bashrc /etc/skel/.profile /root/ && \
-    echo 'alias abort="echo -e '\''q\ny\n'\'' | dlv connect :2345"' >> /root/.bashrc
+RUN echo 'alias abort="echo -e '\''q\ny\n'\'' | dlv connect :2345"' >> /root/.bashrc
 ENV PATH=/var/lib/rancher/rke2/bin:$PATH
 ENV KUBECONFIG=/etc/rancher/rke2/rke2.yaml
 VOLUME /var/lib/rancher/rke2
 # This makes it so we can run and debug k3s too
 VOLUME /var/lib/rancher/k3s
 
-# Dapper/Drone/CI environment
-FROM build AS dapper
-ENV DAPPER_ENV GODEBUG REPO TAG DRONE_TAG PAT_USERNAME PAT_TOKEN KUBERNETES_VERSION DOCKER_BUILDKIT
-ENV DAPPER_OUTPUT ./dist ./bin ./build
-ENV DAPPER_DOCKER_SOCKET true
-ENV DAPPER_TARGET dapper
-ENV DAPPER_RUN_ARGS "-v rke2-pkg:/go/pkg -v rke2-cache:/root/.cache/go-build"
-RUN curl -sL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.27.0
-WORKDIR /source
-# End Dapper stuff
-
 FROM build AS build-k8s
 ARG KUBERNETES_VERSION
 ARG TAG
 WORKDIR /
 RUN git clone -b ${KUBERNETES_VERSION} --depth=1 https://github.com/kubernetes/kubernetes.git
 WORKDIR /kubernetes
-RUN make KUBE_GIT_VERSION=${TAG} GOLDFLAGS='-extldflags "-static"' GOFLAGS='-tags=netgo,osusergo' WHAT='cmd/kube-apiserver cmd/kube-controller-manager cmd/kube-proxy cmd/kube-scheduler cmd/kubeadm cmd/kubectl cmd/kubelet vendor/k8s.io/apiextensions-apiserver'
+RUN echo 'GO_BUILD_FLAGS=" \
+         -mod=vendor \
+         -gcflags=-trimpath=/kubernetes \
+         "' \
+    >> ./go-build-static
+RUN echo 'GO_LDFLAGS=" \
+         -X k8s.io/component-base/version.gitVersion=${TAG} \
+         -X k8s.io/component-base/version.gitCommit=$(git rev-parse HEAD) \
+         -X k8s.io/component-base/version.gitTreeState=clean \
+         -X k8s.io/component-base/version.buildDate=$(date -u +%Y-%m-%dT%H:%M:%SZ) \
```

Comment:
nit: would setup a variable for the date since the two build dates might be off by a second or so. could do the same for git rev.